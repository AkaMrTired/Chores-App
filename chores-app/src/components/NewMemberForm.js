/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from "react";

const NewMemberForm = () => {
  const initialState = {
    fields: {
      role: "",
      email: "",
    },
  };
  const [fields, setFields] = useState(initialState.fields);

  const inviteMember = () => {
    // some axios code to go here to send the field data to the database
    // also nodemailer logic to send email
    // currently the fields will reset but we can change this so we  change the page to be the parent's home page once this is
    setFields(initialState.fields);
  };
  const handleFieldChange = (event) => {
    event.preventDefault();
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="sign-up-container">
      <h1>Invite New Member</h1>
      <div>
        <form onSubmit={inviteMember}>
          <label htmlFor="role">Role</label>
          <select name="role" onChange={handleFieldChange}>
            <option value="">Choose a Role</option>
            <option value="parent">Parent</option>
            <option value="child">Child</option>
          </select>

          <label htmlFor="email">Email</label>
          <input
            type="email"
            name="email"
            required
            placeholder="e.g example@example.com"
            value={fields.email}
            onChange={handleFieldChange}
          />

          <button type="submit">Send Invite!</button>
        </form>
        <button type="button">
          <a href="/dashboard">Back To Dashboard</a>
        </button>
      </div>
    </div>
  );
};

export default NewMemberForm;
