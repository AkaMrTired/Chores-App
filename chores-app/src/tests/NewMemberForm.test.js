import { render, screen } from "@testing-library/react";
import React from "react";
import NewMemberForm from "../components/NewMemberForm";

describe("the new member form", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<NewMemberForm />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<NewMemberForm />);
    expect(screen.getByText(/new member/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /send invite/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/choose a role/i)).toBeInTheDocument();
    expect(screen.getByText(/email/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /back to dashboard/i })
    ).toBeInTheDocument();
  });
});
