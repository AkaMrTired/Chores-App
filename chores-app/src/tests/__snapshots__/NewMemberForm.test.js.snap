// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the new member form matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sign-up-container"
  >
    <h1>
      Invite New Member
    </h1>
    <div>
      <form>
        <label
          for="role"
        >
          Role
        </label>
        <select
          name="role"
        >
          <option
            value="parent"
          >
            Parent
          </option>
          <option
            value="child"
          >
            Child
          </option>
        </select>
        <label
          for="email"
        >
          Email
        </label>
        <input
          name="email"
          placeholder="e.g example@example.com"
          required=""
          type="email"
          value=""
        />
        <button
          type="submit"
        >
          Send Invite!
        </button>
      </form>
      <button
        type="button"
      >
        <a
          href="/dashboard"
        >
          Back To Dashboard
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
